### DISCUSSION TITLE
LGPL license header in files incompatible with Apache v2.0

### DISCUSSION THREAD

**Discussion by rahulmohang** [keycloak/keycloak#19625] [Q&A]
Hi, 

There are license headers for LGPL v2.1 or later in the following files:

https://github.com/keycloak/keycloak/blob/main/federation/sssd/src/main/java/cx/ath/matthew/debug/Debug.java
https://github.com/keycloak/keycloak/blob/main/examples/saml/servlet-filter/src/main/webapp/error.jsp
https://github.com/keycloak/keycloak/blob/main/themes/src/main/resources/theme/keycloak/welcome/index.ftl
https://github.com/keycloak/keycloak/blob/main/examples/saml/servlet-filter/src/main/webapp/logout.jsp

Since LGPL v2.1 or later may be incompatible with the license, Apache v2.0 of the project, do you think that the files introduce a license incompatibility?

**Comment by DAHAG-ArisNourbakhsh** [comment]
https://github.com/keycloak/keycloak/issues/11814

As the Apache License file does not pin the copyright to any valid entity I personally would consider the license to be invalid and not apply to any file that is missing an explicit apache header or has a different licence.  So no conflict is possible /s

Serious answer: AFAIK Apache is compatible with LGPL so there is no issue

**Reply by rahulmohang** [reply]
Thank you for your views. 

As far as I know, and as per the OSADL compatibility matrix, the Apache v2.0 license and LGPL v2.1 or later (when selected LGPL v3.0 or LGPL v3.0 or later) is compatible in one way. i.e. when code licensed under Apache v2.0 gets included in a project licensed under LGPL v3.0 or LGPL v3.0 or later. Inclusion of code licensed under LGPL any version in an Apache v2.0 licensed project may cause license incompatibility. 

In the files listed, I think that the files in example folder may not be a problem as usually examples and test files won't be used in all use scenarios.
